localstack/services/kms/local_kms_provider.py

```diff
@@ -0,0 +1,75 @@
+import logging
+import threading
+from typing import Dict, Optional
+
+from localstack.aws.accounts import get_aws_account_id
+from localstack.aws.api.kms import KmsApi
+from localstack.config import LOCALSTACK_HOSTNAME
+from localstack.constants import DEFAULT_AWS_ACCOUNT_ID
+from localstack.services.infra import log_startup_message
+from localstack.services.kms import local_kms_server
+from localstack.services.plugins import ServiceLifecycleHook
+from localstack.utils.serving import Server
+from localstack.utils.sync import SynchronizedDefaultDict
+
+LOG = logging.getLogger(__name__)
+
+_SERVERS: Dict[str, Server] = {}  # server singleton keyed by account IDs
+_LOCKS = SynchronizedDefaultDict(threading.RLock)
+
+
+class LocalKmsProvider(KmsApi, ServiceLifecycleHook):
+    def get_forward_url(self):
+        """Return the URL of the backend local-kms server to forward requests to"""
+        account_id = get_aws_account_id()
+        start_kms_local(account_id=account_id)
```

Comment:
nit: we should maybe also mention that we're actually starting a server when we're first getting the forward URL.

and some feedback:
overall i think the approach is clever, since the implementation is so simple. but starting servers in property getters is generally something i would avoid. i don't really have a better solution suggestion right now though :-)

Reply:
Thanks, will add that!

_regarding the starting of the server:_
I got that approach from the kinesis-mock implementation, actually. But there it is basically only used as a fallback, not as the only valid implementation in the provider. I reckon implementing a dispatcher which is directly meant for exclusively external services could be something for the future. But in this case we already decided beforehand (in notion) that we want to phase out local-kms mid-term, so I took the quickest approach.

_also, regarding the provider logic:_
As far as I've understood, we would need to add some logic to the pro extensions to make this work when launching with pro, so I switched back to the original separation of providers (again, to choose the quickest approach as I don't need to change anything in pro, it already works and I don't have to change any documentation and do not need a deprecation path). 
If there's some way for a pro-extension to be used on all the available base-providers I'd like to know how to do that. If there isn't, this might be an idea for the future if we plan to continue have multiple implementations of providers in the services.

Reply:
all true :-)